integration-tests/vertx-http/src/test/java/io/quarkus/it/vertx/StaticResourcesTest.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.it.vertx;
+
+import static io.restassured.RestAssured.when;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class StaticResourcesTest {
+
+    @Test
+    public void testExisting() {
+        when().get("/test.txt").then().statusCode(200);
+    }
+
+    @Test
+    public void testNonExisting() {
+        when().get("/test2.txt").then().statusCode(404);
+    }
+
+    @Test
+    public void testIndexInDirectory() {
+        when().get("/dummy/").then().statusCode(200);
+    }
+
+    @Test
+    public void testIndexInNestedDirectory() {
+        when().get("/l1/l2/").then().statusCode(200);
+    }
+
+    @Test
+    public void testNonIndexInDirectory() {
+        when().get("/dummy2/").then().statusCode(404);
```

Comment:
Shouldn't this be included as well?

```suggestion
        when().get("/dummy2/").then().statusCode(200);
```

Reply:
No, because it contains `test.html` not `index.html`

Reply:
Where does this filtering happen? I thought the idea was to register all directories regardless the content (and AFAIU the check for `index.html` is now gone).

Reply:
We register all the directories but the decision of which file is used is still one that is taken by Vertx (and can be configured with a runtime property which defaults to `index.html`)